Formula/coredns.rb

```diff
@@ -0,0 +1,89 @@
+
+class Coredns < Formula
+  desc "DNS server that chains plugins"
+  homepage "https://coredns.io"
+  url "https://github.com/coredns/coredns/archive/v1.0.3.tar.gz"
+  sha256 "dedc42f1a4aac56e397a3fff19955327f21460782aef870658fa1353134ab2f7"
+  head "https://github.com/coredns/coredns.git"
+
+  def default_coredns_config; <<~EOS
+    . {
+      hosts {
+        fallthrough
+      }
+      proxy . 8.8.8.8:53 8.8.4.4:53 {
+        protocol https_google
+      }
+      cache
+      errors
+    }
+    EOS
+  end
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["GOOS"] = "darwin"
+    ENV["GOARCH"] = MacOS.prefer_64_bit? ? "amd64" : "386"
+    ENV.prepend_path "PATH", buildpath/"bin"
+
+    (buildpath/"src/github.com/coredns/coredns").install buildpath.children
+
+    cd "src/github.com/coredns/coredns" do
+      # Replace git call with version
+      inreplace "Makefile", "$(shell git describe --dirty --always)", "v1.0.3"
```

Comment:
Could always just build from the git checkout. If not does something like `make coredns GITCOMMIT=#{version}` work?